examples/README.md

```diff
@@ -267,3 +278,73 @@ Add the test runner to the scripts in your package.json
 ```
 
 Run your tests with `yarn test`
+
+# Using code examples in the docs site
+
+Creating and using code snippets is relatively straight forward. You can pull in a whole file or by adding some comments you can pull in just a particular few lines of a file as necessary.
+
+To streamline the PR process, you should create a branch in the `docs` repo to add your code snippets before your branch for the code example in `ts-immutable-sdk` is merged to main. This way we can point the docs site to your branch for the code snippets and you can see your code example in situ and make any changes required to your code example before going through the PR process.
+
+## Update import code branch
+
+In your `docs` branch modify the file `/src/remark/import-code.mjs` and update the `BRANCH` constant from `main` to the name of your branch on `ts-immutable-sdk` e.g.
+
+```
+const BRANCH = 'DVR-193-passport-signing-example';
+```
+
+Now your `docs` branch will be pulling the code examples from your branch in `ts-immutable-sdk` and you can use them locally in your `docs` branch to make sure they make sense in the page.
+
+Make sure to change this back to `main` before merging your `docs` branch or it will break the docs site.
+
+## Creating code snippets
+
+In your code examples in `tsimmutable-sdk` find the parts of the code you want to use as code snippets and wrap them in the `#doc` and `#enddoc` comments while providing a label.
+
+Labels have to be unique within a file so you should use a simple naming convention to avoid clashes within the file e.g.
+
+e.g. `<library>-<tag>`
+
+```
+// #doc eip1193-create
+const passportProvider = passportInstance.connectEvm()
+// #enddoc eip1193-create
+```
+
+Make sure to commit and push the labels to your `ts-immutable-sdk` branch on GitHub so they can be pulled down into your `docs` local build from there.
+
+## Using code snippets
+
+It's very simple to use the code snippet in the docs site, you just add a code block with the reference to the file and the `#` of the label you want to display e.g.
+
+````
+```tsx reference=examples/passport/wallets-connect-with-nextjs/src/app/connect-with-eip1193/page.tsx#eip1193-create title="Create the Passport Provider"
+```
+````
+Or if you want to display the whole file just don't include a `#` label at the end of the file reference e.g.
+
+````
+```tsx reference=examples/passport/wallets-connect-with-nextjs/src/app/connect-with-eip1193/page.tsx title="Create the Passport Provider"
+```
+````
+
+Just like regular code snippets, you can set language for syntax highlighting, in this example we are setting it to `tsx`.
```

Comment:
It's not clear to me what languages this supports out of the box. Is there a link we can refer to or add another row in the table below? Not a biggie though as I don't imagine this edge case will be encountered often